src/side-sheet/src/SheetClose.js

```diff
@@ -11,72 +12,139 @@ const animationEasing = {
+      css.keyframes('rotate360InAnimation', {
+        from: { transform: `translateX(100%) rotate(0deg)` },
+        to: { transform: `translateX(-100%) rotate(-360deg)` }
+      }),
+      css.keyframes('rotate360OutAnimation', {
+        from: { transform: `translateX(-100%) rotate(0deg)` },
+        to: { transform: `translateX(100%) rotate(360deg)` }
+      })
+    )
   },
-  '&[data-state="entering"], &[data-state="entered"]': {
-    animation: `${rotate360InAnimation} ${ANIMATION_DURATION}ms ${
-      animationEasing.deceleration
-    } both`
+  [Position.LEFT]: {
+    marginRight: -12,
+    right: 0,
+    marginTop: 12,
+    transform: `translateX(100%)`,
+    ...withAnimations(
+      css.keyframes('leftRotate360InAnimation', {
+        from: { transform: `translateX(-100%) rotate(0deg)` },
+        to: { transform: `translateX(100%), rotate(360deg)` }
+      }),
+      css.keyframes('leftRotate360OutAnimation', {
+        from: { transform: `translateX(100%) rotate(0deg)` },
+        to: { transform: `translateX(-100%), rotate(360deg)` }
+      })
+    )
+  },
+  [Position.TOP]: {
+    right: 0,
+    marginRight: 12,
+    top: '100%',
+    marginTop: 12,
+    transform: `translateY(0)`,
+    ...withAnimations(
+      css.keyframes('topRotate360InAnimation', {
+        from: { transform: `translateY(-200%) rotate(0deg)` },
+        to: { transform: `translateY(0%), rotate(360deg)` }
+      }),
+      css.keyframes('topRotate360OutAnimation', {
+        from: { transform: `translateY(0%) rotate(0deg)` },
+        to: { transform: `translateY(-200%), rotate(360deg)` }
+      })
+    )
   },
-  '&[data-state="exiting"]': {
-    animation: `${rotate360OutAnimation} ${ANIMATION_DURATION}ms ${
-      animationEasing.acceleration
-    } both`
+  [Position.BOTTOM]: {
+    right: 0,
+    marginRight: 12,
+    bottom: '100%',
+    marginBottom: 12,
+    transform: `translateY(0)`,
+    ...withAnimations(
+      css.keyframes('topRotate360InAnimation', {
+        from: { transform: `translateY(200%) rotate(0deg)` },
+        to: { transform: `translateY(0%), rotate(360deg)` }
+      }),
+      css.keyframes('topRotate360OutAnimation', {
+        from: { transform: `translateY(0%) rotate(0deg)` },
+        to: { transform: `translateY(200%), rotate(360deg)` }
+      })
+    )
   }
-})
+}
+
+const sheetCloseClassNameCache = {}
+
+const getSheetCloseClassName = position => {
+  if (!sheetCloseClassNameCache[position]) {
+    sheetCloseClassNameCache[position] = css({
+      ...sheetCloseStyles[position],
+      ...sharedStyles
+    }).toString()
+  }
+  return sheetCloseClassNameCache[position]
+}
 
 export default class SheetClose extends PureComponent {
   static propTypes = {
     ...Box.propTypes,
-    isClosing: PropTypes.bool
-  }
-
-  static styles = {
-    position: 'absolute',
-    marginLeft: -12,
-    marginTop: 12,
-    padding: 4,
-    borderRadius: 9999
+    isClosing: PropTypes.bool,
+    position: PropTypes.oneOf([
```

Comment:
Do we need a default position? or else mark this as required?

Reply:
I think we should probably just mark it as required. The default value is effectively provided by the SideSheet component.